<?xml version="1.0"?>

<robot name="scara">

  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://scara_HS-045/meshes/base_link.dae"/>
      </geometry>      
    </visual>
  </link>

  <link name="link_1">
  	<visual>
      <geometry>
        <mesh filename="package://scara_HS-045/meshes/link1.dae"/>
      </geometry>      
    </visual>
  </link>

  <link name="link_2">
  	<visual>
      <geometry>
        <mesh filename="package://scara_HS-045/meshes/link2.dae"/>
      </geometry>      
    </visual>
  </link>

  <link name="link_3">
  	<visual>
      <geometry>
        <mesh filename="package://scara_HS-045/meshes/link3.dae"/>
      </geometry>      
    </visual>
  </link>
  
  <link name="tool" />
  <link name="end" />

  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link_1"/>
    <origin xyz="0 0 0.301"/>
    <axis xyz="0 0 1" />
    <limit effort="300" velocity="0.1" lower="-1.57" upper="1.57"/>
  </joint>

  <joint name="joint2" type="revolute">
    <parent link="link_1"/>
    <child link="link_2"/>
    <origin xyz="0.225 0 0.100"/>
    <axis xyz="0 0 1" />
    <limit effort="300" velocity="0.1" lower="-1" upper="1"/>
  </joint>

  <joint name="joint3" type="prismatic">
    <parent link="link_2"/>
    <child link="link_3"/>
    <origin xyz="0.225 0 0"/>
    <axis xyz="0 0 1" />
    <limit effort="300" velocity="0.1" lower="-0.205" upper="0"/>
  </joint>

  <joint name="joint4" type="revolute">
    <parent link="link_3"/>
    <child link="end"/>
    <origin xyz="0 0 -0.194"/>
    <axis xyz="0 0 1" />
    <limit effort="300" velocity="0.1" lower="-4.71" upper="4.71"/>
    
  </joint>

  <joint name="tool_to_link3" type="fixed">
    <parent link="end"/>
    <child link="tool"/>
    <origin xyz="0 0 0"/>
   
  </joint>    

</robot>